Sources/CacheContainer.swift

```diff
@@ -46,3 +46,68 @@ public final class CacheContainer {
         }
     }
 }
+
+final class ThreadSafeDictionary<V: Hashable,T>: Collection {
```

Comment:
Why is this type a `class` and not a `struct`?

Reply:
When I use `ThreadSafeDictionary` with `struct`, I get the error `Escaping closure captures mutating 'self' parameter` in `subscript` and `removeValue`.
It seems difficult to update `self` in a `struct` closure, so I use `class`.

```swift
struct ThreadSafeDictionary<V: Hashable,T>: Collection {
    private var dictionary: [V: T]
...
    subscript(key: V) -> T? {
        set(newValue) {
            self.concurrentQueue.async(flags: .barrier) {
                dictionary[key] = newValue   //   Error occurred: Escaping closure captures mutating 'self' parameter
            }
        }
        get {
            self.concurrentQueue.sync {
                return self.dictionary[key]
            }
        }
    }

...
    mutating func removeValue(forKey key: V) {
        self.concurrentQueue.async(flags: .barrier) {
            dictionary.removeValue(forKey: key)   //   Error occurred: Escaping closure captures mutating 'self' parameter
        }
    }
}
```

Reply:
Thank you for your explanation. Understood.